pkg/repositories/repositories.go

```diff
@@ -58,6 +58,41 @@ func NewConfig(filename string) (Config, error) {
 	return repoConfig, nil
 }
 
+// UpdateConfigVersions updates a config in-place, setting the afterVersion to
+// its version, and it's version to a newer one specified in a secondary config
+func UpdateConfigVersions(baseConfig *Config, configWithNewVersions *Config) {
```

Comment:
I think this method belongs to baseConfig. It wouldn't make sense without it.
With that in mind, perhaps we can change the signature to 
```
func (baseConfig *Config) UpdateConfigVersions(configWithNewVersions *Config) {
```

Reply:
I think the name `UpdateConfigVersions` is a bit ambiguous and so doesn't quite capturing the sense of what the function is doing. I think more precisely the function is upgrading repo versions so maybe it's better named `UpgradeRepoVersions`. I like Upgrade more than Update because it makes explicit the idea that we're going from lower to higher versions.

What, if anything, shall we do when `newVersion < currentRepo.Version` ?
```
remappedRepo.AfterVersion = currentRepo.Version
remappedRepo.Version = newVersion
````

Reply:
Since the function is doing a few things it might be worth splitting the description into a few sentences e.g.
```
// UpdateRepos updates a config's repos' versions in-place, if the repo has a newer
// version in the supplied config. It does this through the following steps:
// 1. Sets the 'afterVersion' to the current 'version' of a repo
// 2. Updates the 'version' to the newer one
```

Reply:
I disagree with the change regarding the name being 'upgrade' vs 'update', as the functionality is not an upgrade. We're changing both the version and the afterversion, not necessarily upgrading one or both. If we were just changing one, I'd agree that it's an upgrade, otherwise I think it's a bit misleading.

Reply:
@BradleyBoutcher I thought we were necessarily upgrade given the semantics of afterVersion and version are dictate  version >= afterVersion, since afterVersion is the lower bound. That's also why I asked 
> What, if anything, shall we do when newVersion < currentRepo.Version ?

I'm also wondering what should happen when the config with new versions specifies both version and afterVersion.

Reply:
Regardless of the verb we go for I think `XRepoVersions` is a more accurate description of the functionality than `XConfigVersions`.